File: app/src/main/java/com/ngengs/android/popularmovies/apps/utils/GridSpacesItemDecoration.java
package com.ngengs.android.popularmovies.apps.utils;

import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * Created by ngengs on 6/16/2017.
 */

@SuppressWarnings("DefaultFileTemplate")
public class GridSpacesItemDecoration extends RecyclerView.ItemDecoration {
    private final int mSpanCount;
    private final int mSpacing;


    public GridSpacesItemDecoration(int spanCount, int spacing) {
        this.mSpanCount = spanCount;
        this.mSpacing = spacing;
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent,
                               RecyclerView.State state) {
        int position = parent.getChildAdapterPosition(view);
        int column = position % mSpanCount;

        outRect.left = mSpacing - column * mSpacing / mSpanCount;
        outRect.right = (column + 1) * mSpacing / mSpanCount;

        if (position < mSpanCount) outRect.top = mSpacing;

        outRect.bottom = mSpacing;
    }
}

File: app/src/main/java/com/ngengs/android/popularmovies/apps/utils/GridSpacesItemDecoration.kt
package com.ngengs.android.popularmovies.apps.utils

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by rizky.kharisma on 09/11/22.
 * @ngengs
 */
class GridSpacesItemDecoration(
    private val spanCount: Int,
    private val spacing: Int,
): RecyclerView.ItemDecoration() {

    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: RecyclerView.State
    ) {
        val position = parent.getChildAdapterPosition(view)
        val column = position % spanCount

        outRect.left = spacing - column * spacing / spanCount
        outRect.right = (column + 1) * spacing / spanCount

        if (position < spanCount) outRect.top = spacing

        outRect.bottom = spacing
    }
}